File: app/src/main/java/in/ceeq/alec/network/ContactService.java
package in.ceeq.alec.network;

import java.util.List;

import in.ceeq.alec.databinding.Contact;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface ContactService {

    @GET("users")
    Call<List<Contact>> getContacts();

    @Headers({
            "Accept: application/json",
            "Content-Type: application/json"
    })
    @GET("users/{id}")
    Call<Contact> getContact(@Path("id") int id);

    @Headers({
            "Accept: application/json",
            "Content-Type: application/json"
    })
    @POST("users")
    Call<Contact> postContact(@Body Contact contact);

    @Headers({
            "Accept: application/json",
            "Content-Type: application/json"
    })
    @PUT("users/{id}")
    Call<Contact> putContact(@Body Contact contact, @Path("id") int id);
}

File: app/src/main/java/in/ceeq/lyte/network/ContactService.kt
package `in`.ceeq.lyte.network

import `in`.ceeq.lyte.databinding.Contact
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface ContactService {

    @get:GET("users")
    val contacts: Call<List<Contact>>

    @Headers("Accept: application/json", "Content-Type: application/json")
    @GET("users/{id}")
    fun getContact(@Path("id") id: Int): Call<Contact>

    @Headers("Accept: application/json", "Content-Type: application/json")
    @POST("users")
    fun postContact(@Body contact: Contact): Call<Contact>

    @Headers("Accept: application/json", "Content-Type: application/json")
    @PUT("users/{id}")
    fun putContact(@Body contact: Contact, @Path("id") id: Int): Call<Contact>
}
